vm/src/obj/objtype.rs

```diff
@@ -85,16 +97,59 @@ pub fn base_classes(obj: &PyObjectRef) -> Vec<PyObjectRef> {
     _mro(obj.typ()).unwrap()
 }
 
-pub fn isinstance(obj: &PyObjectRef, cls: &PyObjectRef) -> bool {
+/// Determines if obj actually an instance of cls, this doesn't call __instancecheck__, so only use
+/// this if cls is known to have not overridden the base __instancecheck__ magic method.
+pub fn real_isinstance(obj: &PyObjectRef, cls: &PyObjectRef) -> bool {
```

Comment:
My suggestion would be to leave the original name unchanged. Actually `real_isinstance` more sounds like the fully fledged thing that also invokes metaclass magic methods if they are present.